### DISCUSSION TITLE
Flask REST and Flask Socket io

### DISCUSSION THREAD

**Discussion by sree9578** [miguelgrinberg/Flask-SocketIO#1773] [Q&A] [answered]
i tried to emit the response via sockets when there is request in api calls. I have multiprocessed the app and sockets in different ports. i am using postman to check the response . i am able to get response in postman when the socketio.on connect works...but under route the emit is not reflected to the postman..i don't know whether it is a port issue or due to threading ..can u please provide a solution to this problem as soon as possible.
@miguelgrinberg   thanks in advance

```
###server file
from flask import Flask
from gevent import monkey
monkey.patch_all()
from flask_socketio import SocketIO, emit

app = Flask(__name__)

socketio = SocketIO(app,ping_timeout=80,thread=True)

@app.route('/rap', methods=['POST'])
def  check():
...
..
..
socketio.emit('used', {'data': 'Connected'},broadcast=True)
```

i tried with namespace also the message is not emitted in postman

```
def run_sockt():
   socketio.run(app,host='0.0.0.0',port=6677,debug=False,use_reloader=False)


def run_app():
   app.run(host='0.0.0.0',port=8899,threaded=True,use_reloader=False)

if __name__ == '__main__':
   
   
   print('nooo')

   p1=multiprocessing.Process(target=run_app)
   p2=multiprocessing.Process(target=run_sockt)
   p1.start()
   p2.start()
   p1.join()
   p2.join()
```

**Comment by miguelgrinberg** [comment, marked as answer]
@sudhan-c please do not write bug reports on this project unless you have a bug to report. Asking why your application doesn't work doesn't qualify as a bug report. That is what the discussion board is for.

Regarding your problem, you cannot use a `socketio` instance in across processes. See the documentation regarding how to emit from an auxiliary process.

**Reply by sree9578** [reply]
Sorry will do it correctly from next time...can u please guide me on the part of how to emit from an auxiliary process..Thanks @miguelgrinberg

**Reply by sree9578** [reply]
In this case i am using socketio as a notification sender for POST request calls..

**Reply by miguelgrinberg** [reply]
Read the documentation please. https://flask-socketio.readthedocs.io/en/latest/deployment.html#emitting-from-an-external-process
